api/docs/org.jboss.tools.rsp.schema/src/main/java/org/jboss/tools/rsp/api/schema/TypescriptUtility.java

```diff
@@ -120,4 +140,340 @@ private static String safeReadFile(Path p) {
+		for( AnnotationExpr a : annotations) {
+			String annotName = annotations.get(0).getNameAsString();
+			if( annotName.equalsIgnoreCase("JsonRequest")) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	
+	private void printOneRequest(String methodName, JavadocComment jdc, 
+			StringBuffer sb, String serverOrClient) {
+		if( jdc != null ) {
+			String comment = jdc.getContent().substring(1);
+			sb.append("        /**\n");
+			sb.append(comment.replaceAll("\t", "        "));
+			sb.append("*/");
+			sb.append("\n        export namespace ");
+			sb.append(methodName.substring(0, 1).toUpperCase());
+			sb.append(methodName.substring(1));
+			sb.append("Request {\n");
+			
+			// TODO body
+			MethodDeclaration md = getMethodDeclaration(jdc);
+			sb.append("            export const type = new RequestType<");
+			NodeList<Parameter> params = md.getParameters();
+			if( params.size() == 0 ) {
+				sb.append("void, ");
+			} else {
+				Type type = params.get(0).getType();
+				String typeName = type.toString();
+				if( typeName.equalsIgnoreCase("void")) 
+					sb.append("void, ");
+				else 
+					sb.append("Protocol." + typeName + ", ");
+
+			}
+			
+			Type retType = md.getType();
+			String retTypeName = convertReturnType(retType.toString());
+			sb.append(retTypeName);
+			sb.append(", void, void>('" + serverOrClient + "/" + methodName + "');");
+			sb.append("\n");
+			sb.append("        }\n");
+		}
+	}
+
+	private void printOneNotification(String methodName, JavadocComment jdc, 
+			StringBuffer sb, String serverOrClient) {
+		if( jdc != null ) {
+			String comment = jdc.getContent().substring(1);
+			sb.append("        /**\n");
+			sb.append(comment.replaceAll("\t", "        "));
+			sb.append("*/");
+			sb.append("\n        export namespace ");
+			sb.append(methodName.substring(0, 1).toUpperCase());
+			sb.append(methodName.substring(1));
+			sb.append("Notification {\n");
+			
+			// TODO body
+			MethodDeclaration md = getMethodDeclaration(jdc);
+			sb.append("            export const type = new NotificationType<");
+			NodeList<Parameter> params = md.getParameters();
+			if( params.size() == 0 ) {
+				sb.append("void, ");
+			} else {
+				Type type = params.get(0).getType();
+				String typeName = type.toString();
+				if( typeName.equalsIgnoreCase("void")) 
+					sb.append("void, ");
+				else 
+					sb.append("Protocol." + typeName + ", ");
+			}
+			
+			Type retType = md.getType();
+			String retTypeName = convertReturnType(retType.toString());
+			sb.append(retTypeName);
+			sb.append(">('" + serverOrClient + "/" + methodName + "');");
+			sb.append("\n");
+			sb.append("        }\n");
+		}
+	}
+
+	private String convertReturnType(String type) {
+		if( type.startsWith("CompletableFuture<") && type.endsWith(">")) {
+			type = type.substring("CompletableFuture<".length());
+			type = type.substring(0, type.length()-1);
+		}
+		if( type.startsWith("List<")) {
+			type = "Array<Protocol." + type.substring("List<".length());
```

Comment:
Can you change ```Array<Type>``` into ```Type[]```? TSLint keeps bitching the generic array is not the correct convention.